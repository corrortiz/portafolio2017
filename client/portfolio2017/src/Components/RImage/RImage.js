/* eslint-disable */
// @flow
import React, { Component } from 'react';

/**
 * Component to receive images and be responsive
 */
class RImage extends Component {
  render() {
    const {
      lg_1x,
      lg_2x,
      md_1x,
      md_2x,
      sm_1x,
      sm_2x,
      lg_1x_JPG,
      md_1x_JPG,
      sm_1x_JPG
    } = this.props;

    return (
      <picture>
        <source
          media="(min-width: 900px)"
          srcSet={`${lg_1x} 1x, ${lg_2x} 2x`}
          type="image/webp"
        />
        <source
          media="(min-width: 601px)"
          srcSet={`${md_1x} 1x, ${md_2x} 2x`}
          type="image/webp"
        />
        <source srcSet={`${sm_1x} 1x, ${sm_2x} 2x`} type="image/webp" />
        <img
          srcSet={`${sm_1x_JPG} 600w,
            ${md_1x_JPG} 900w,
            ${lg_1x_JPG} 1440w`}
          src={lg_1x_JPG}
          type="image/jpeg"
          alt="image description"
        />
      </picture>
    );
  }
}

export default RImage;
