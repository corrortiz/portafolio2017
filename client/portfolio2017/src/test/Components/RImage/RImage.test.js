import React from 'react';
import { shallow } from 'enzyme';
import RImage from '../../../Components/RImage/RImage';

describe('RImage Component', () => {
  it('should compara a snapshot of RImage', () => {
    const wrapper = shallow(<RImage />);
    expect(wrapper).toMatchSnapshot();
  });
});
