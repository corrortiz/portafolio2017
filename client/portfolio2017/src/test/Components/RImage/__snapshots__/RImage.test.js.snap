// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RImage Component should compara a snapshot of RImage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RImage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <source
          media="(min-width: 900px)"
          srcSet="undefined 1x, undefined 2x"
          type="image/webp"
/>,
        <source
          media="(min-width: 601px)"
          srcSet="undefined 1x, undefined 2x"
          type="image/webp"
/>,
        <source
          srcSet="undefined 1x, undefined 2x"
          type="image/webp"
/>,
        <img
          alt="image description"
          src={undefined}
          srcSet="undefined 600w,
            undefined 900w,
            undefined 1440w"
          type="image/jpeg"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "media": "(min-width: 900px)",
          "srcSet": "undefined 1x, undefined 2x",
          "type": "image/webp",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "media": "(min-width: 601px)",
          "srcSet": "undefined 1x, undefined 2x",
          "type": "image/webp",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "srcSet": "undefined 1x, undefined 2x",
          "type": "image/webp",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "image description",
          "src": undefined,
          "srcSet": "undefined 600w,
            undefined 900w,
            undefined 1440w",
          "type": "image/jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "picture",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <source
            media="(min-width: 900px)"
            srcSet="undefined 1x, undefined 2x"
            type="image/webp"
/>,
          <source
            media="(min-width: 601px)"
            srcSet="undefined 1x, undefined 2x"
            type="image/webp"
/>,
          <source
            srcSet="undefined 1x, undefined 2x"
            type="image/webp"
/>,
          <img
            alt="image description"
            src={undefined}
            srcSet="undefined 600w,
            undefined 900w,
            undefined 1440w"
            type="image/jpeg"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "media": "(min-width: 900px)",
            "srcSet": "undefined 1x, undefined 2x",
            "type": "image/webp",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "media": "(min-width: 601px)",
            "srcSet": "undefined 1x, undefined 2x",
            "type": "image/webp",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "srcSet": "undefined 1x, undefined 2x",
            "type": "image/webp",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "image description",
            "src": undefined,
            "srcSet": "undefined 600w,
            undefined 900w,
            undefined 1440w",
            "type": "image/jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "picture",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
